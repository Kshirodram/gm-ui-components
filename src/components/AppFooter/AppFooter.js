import React from "react";
import PropTypes from "prop-types";

import Copyright from "./components/Copyright";
import Footer from "./components/Footer";
import Links from "./components/Links";
import Link from "./components/Link";
import { IconTwitter, IconGitHub, IconLinkedIn } from "components/Glyphs";
import LongLogo from "./components/LongLogo";

import longLogo from "./assets/decipher-logo-long.png";

AppFooter.propTypes = {
  copyrightText: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string
  ]),
  useExternalLinks: PropTypes.bool
};

AppFooter.defaultProps = {
  copyrightText:
    "Copyright \u00A9 2018 Decipher Technology Studios. All rights reserved. \n Copyright \u00A9 2018 Grey Matter, a Decipher Technology Studios product. All rights reserved.",
  useExternalLinks: true
};

/**
 * Stateless functional React component that renders company branding and social media footer content
 * @returns JSX.Element
 */
export default function AppFooter({
  copyrightText,
  useExternalLinks,
  ...props
}) {
  const logoProps = useExternalLinks
    ? {
        href: "http://deciphernow.com",
        rel: "noopener noreferrer",
        target: "_blank"
      }
    : {};
  return (
    <Footer {...props}>
      <LongLogo {...logoProps} title="Decipher Technology Studios website">
        <img alt="Decipher Technology Studios" src={longLogo} />
      </LongLogo>
      <Copyright useExternalLinks={useExternalLinks}>
        {" "}
        {copyrightText}{" "}
      </Copyright>
      {useExternalLinks && (
        <Links>
          <Link
            href="http://github.com/DecipherNow"
            rel="noopener noreferrer"
            target="_blank"
            title="Decipher Technology Studios Github"
          >
            <IconGitHub size="20px" />
          </Link>
          <Link
            href="http://twitter.com/deciphernow"
            rel="noopener noreferrer"
            target="_blank"
            title="Decipher Technology Studios Twitter"
          >
            <IconTwitter size="20px" />
          </Link>
          <Link
            href="http://www.linkedin.com/company/decipher-technology-studios"
            rel="noopener noreferrer"
            target="_blank"
            title="Decipher Technology Studios LinkedIn"
          >
            <IconLinkedIn size="20px" />
          </Link>
        </Links>
      )}
    </Footer>
  );
}

AppFooter.displayName = "AppFooter";
