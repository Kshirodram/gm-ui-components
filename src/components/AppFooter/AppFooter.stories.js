import React from "react";
import { storiesOf } from "@storybook/react";
import { boolean, text } from "@storybook/addon-knobs/react";
import AppFooter from "./AppFooter";
import { withInfo } from "@storybook/addon-info";

const stories = storiesOf("Components|App Footer", module);

stories.add(
  "default",
  withInfo("AppFooter for Grey Matter products.")(() => (
    <AppFooter
      copyrightText={text("copyrightText")}
      useExternalLinks={boolean("useExternalLinks", true)}
    />
  ))
);
