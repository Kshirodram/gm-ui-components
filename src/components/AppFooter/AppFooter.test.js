import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import AppFooter from "./AppFooter.js";

describe("A Footer container", () => {
  it("should render", () => {
    const anAppFooter = renderer.create(<AppFooter />).toJSON();
    expect(anAppFooter).toMatchSnapshot();
  });
});

describe("useExternalLinks is true", () => {
  const anAppFooter = shallow(<AppFooter useExternalLinks={true} />);

  it("useExternalLinks boolean prop is true", () => {
    expect(anAppFooter.prop.useExternalLinks).toBeTruthy;
  });

  it("disables logo link", () => {
    const footerHtml = anAppFooter.find("LongLogo").html();
    expect(footerHtml.includes("href")).toBeTruthy;
    expect(footerHtml.includes("rel")).toBeTruthy;
    expect(footerHtml.includes("target")).toBeTruthy;
  });

  it("hides social links", () => {
    expect(anAppFooter.find("Links")).toBeTruthy;
  });
});

describe("useExternalLinks is false", () => {
  const anAppFooter = shallow(<AppFooter useExternalLinks={false} />);

  it("useExternalLinks boolean prop is false", () => {
    expect(anAppFooter.prop.useExternalLinks).toBeFalsy;
  });

  it("disables logo link", () => {
    const footerHtml = anAppFooter.find("LongLogo").html();
    expect(footerHtml.includes("href")).toBeFalsy;
    expect(footerHtml.includes("rel")).toBeFalsy;
    expect(footerHtml.includes("target")).toBeFalsy;
  });

  it("hides social links", () => {
    expect(anAppFooter.find("Links")).toBeFalsy;
  });
});
