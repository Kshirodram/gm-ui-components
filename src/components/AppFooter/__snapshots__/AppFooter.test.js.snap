// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Footer container should render 1`] = `
.c2 {
  font-size: 50%;
  margin: 0;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  text-align: center;
  -webkit-transition: opacity 0.2s ease;
  transition: opacity 0.2s ease;
  cursor: default;
  border-top: 1px solid;
  margin-top: 8px;
  padding: 8px;
  text-transform: uppercase;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  white-space: pre-wrap;
}

.c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  padding: 0 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  height: auto;
  min-height: 32px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

.c3 {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  text-align: right;
  padding: 0 0 4px 8px;
}

.c4 {
  padding: 0 4px;
}

.c4 svg {
  height: 18px;
}

.c1 {
  padding: 0 8px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  text-align: left;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-filter: grayscale(100%);
  filter: grayscale(100%);
}

.c1:hover {
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-filter: grayscale(0%);
  filter: grayscale(0%);
}

.c1 img {
  max-width: 140px;
  vertical-align: -3px;
}

@media all and (min-width:800px) {
  .c2 {
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    padding: 0;
    margin: 0;
    border: none;
  }
}

@media all and (min-width:800px) {
  .c0 {
    height: 40px;
    -webkit-flex-wrap: no-wrap;
    -ms-flex-wrap: no-wrap;
    flex-wrap: no-wrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
  }

  .c0 + * {
    padding-bottom: 32px;
  }
}

@media all and (min-width:800px) {
  .c3 {
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
  }
}

<footer
  className="c0"
>
  <a
    className="c1"
    href="http://deciphernow.com"
    rel="noopener noreferrer"
    target="_blank"
    title="Decipher Technology Studios website"
  >
    <img
      alt="Decipher Technology Studios"
      src="decipher-logo-long.png"
    />
  </a>
  <p
    className="c2"
  >
     
    Copyright © 2018 Decipher Technology Studios. All rights reserved. 
 Copyright © 2018 Grey Matter, a Decipher Technology Studios product. All rights reserved.
     
  </p>
  <div
    className="c3"
  >
    <a
      className="c4"
      href="http://github.com/DecipherNow"
      rel="noopener noreferrer"
      target="_blank"
      title="Decipher Technology Studios Github"
    >
      <IconMock
        size="20px"
      />
    </a>
    <a
      className="c4"
      href="http://twitter.com/deciphernow"
      rel="noopener noreferrer"
      target="_blank"
      title="Decipher Technology Studios Twitter"
    >
      <IconMock
        size="20px"
      />
    </a>
    <a
      className="c4"
      href="http://www.linkedin.com/company/decipher-technology-studios"
      rel="noopener noreferrer"
      target="_blank"
      title="Decipher Technology Studios LinkedIn"
    >
      <IconMock
        size="20px"
      />
    </a>
  </div>
</footer>
`;
